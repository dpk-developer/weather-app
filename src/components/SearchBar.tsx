import React from 'react';
import {
  View,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  Image,
} from 'react-native';

import { useTheme } from '@hooks/useTheme';

interface Props {
  city: string;
  onChange: (t: string) => void;
  onSubmit: () => void;
}

const IC_SEARCH = require('@assets/icons/search.png');

const SearchBar = ({ city, onChange, onSubmit }: Props) => {
  const { theme } = useTheme();

  return (
    <View style={[styles.container, { backgroundColor: theme.background }]}>
      <TextInput
        style={[
          styles.input,
          {
            color: theme.text,
            backgroundColor: theme.inputBackground,
          },
        ]}
        placeholder="Enter city"
        placeholderTextColor={theme.placeholder}
        value={city}
        onChangeText={onChange}
      />
      <TouchableOpacity
        accessibilityRole="button"
        testID="search-button"
        style={[styles.iconContainer, { backgroundColor: theme.button }]}
        onPress={onSubmit}>
        <Image
          source={IC_SEARCH}
          style={[styles.searchIcon, { tintColor: theme.icon }]}
        />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: 'white',
    borderRadius: 8,
    elevation: 5,
    shadowColor: '#000',
    shadowOpacity: 0.1,
    shadowOffset: { width: 0, height: 3 },
    shadowRadius: 4,
    margin: 16,
    height: 54,
  },
  input: {
    flex: 1,
    fontSize: 16,
    paddingLeft: 8,
    fontWeight: '500',
    borderTopLeftRadius: 8,
    borderBottomLeftRadius: 8,
  },
  iconContainer: {
    width: 60,
    alignItems: 'center',
    justifyContent: 'center',
    borderTopRightRadius: 8,
    borderBottomRightRadius: 8,
  },
  searchIcon: {
    width: 24,
    height: 24,
  },
});

export default SearchBar;
