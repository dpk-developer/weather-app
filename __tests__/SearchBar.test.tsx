import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import SearchBar from '../src/components/SearchBar';

jest.mock('../src/hooks/useTheme.ts', () => ({
  useTheme: () => ({
    theme: {
      background: '#fff',
      text: '#000',
      inputBackground: '#eee',
      placeholder: '#888',
      button: '#333',
      icon: '#fff',
    },
  }),
}));

describe('SearchBar', () => {
  it('renders correctly and handles input', () => {
    const mockChange = jest.fn();
    const mockSubmit = jest.fn();

    const { getByPlaceholderText, getByTestId, toJSON } = render(
      <SearchBar city="London" onChange={mockChange} onSubmit={mockSubmit} />,
    );

    const input = getByPlaceholderText('Enter city');
    fireEvent.changeText(input, 'Paris');

    const button = getByTestId('search-button');
    fireEvent.press(button);

    expect(mockChange).toHaveBeenCalledWith('Paris');
    expect(mockSubmit).toHaveBeenCalled();
    expect(toJSON()).toMatchSnapshot();
  });
});
