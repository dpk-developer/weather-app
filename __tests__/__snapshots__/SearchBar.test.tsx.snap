// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly and handles input 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 8,
        "elevation": 5,
        "flexDirection": "row",
        "height": 54,
        "margin": 16,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 4,
      },
      {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <TextInput
    onChangeText={
      [MockFunction] {
        "calls": [
          [
            "Paris",
          ],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    placeholder="Enter city"
    placeholderTextColor="#888"
    style={
      [
        {
          "borderBottomLeftRadius": 8,
          "borderTopLeftRadius": 8,
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "500",
          "paddingLeft": 8,
        },
        {
          "backgroundColor": "#eee",
          "color": "#000",
        },
      ]
    }
    value="London"
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#333",
        "borderBottomRightRadius": 8,
        "borderTopRightRadius": 8,
        "justifyContent": "center",
        "opacity": 1,
        "width": 60,
      }
    }
    testID="search-button"
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/icons/search.png",
        }
      }
      style={
        [
          {
            "height": 24,
            "width": 24,
          },
          {
            "tintColor": "#fff",
          },
        ]
      }
    />
  </View>
</View>
`;
